pkg/istio-agent/agent.go

```diff
@@ -658,7 +659,7 @@ func (a *Agent) newSecretManager() (*cache.SecretManagerClient, error) {
 			log.Infof("Using CA %s cert with certs: %s", a.secOpts.CAEndpoint, caCertFile)
 		}
 	}
-
+	a.secOpts.CertSigner = a.proxyConfig.ProxyMetadata[MetadataCertSigner]
```

Comment:
ProxyMetadata becomes an environment variable. We should just read from the environment in pilot/cmd/pilot-agent/options/options.go

Reply:
Thanks! fixed

Reply:
Thanks! fixed